R/updateNullModOutcome.R

```diff
@@ -65,57 +63,64 @@ updateNullModOutcome <- function(nullmod,
         }
 
         # rank normalize and rescale
-        resid <- .rankNorm(resid) 
-        resid <- resid*sqrt(all.vars)    
-    } 
+        resid <- .rankNorm(resid)
+        resid <- resid*sqrt(all.vars)
+    }
 
     # old code for rankNorm.option == "all"
     # for (i in 1:g){
     #     resid[group.idx[[i]]] <- resid[group.idx[[i]]]*sqrt(group.vars[i])/sd(resid[group.idx[[i]]])
     # }
-    
+
     ### now re-fit the null model:
-    
+
     new.nullmod <- .fitNullModel(y = resid, X = nullmod$model.matrix, covMatList = covMatList,
-                                 group.idx = group.idx, family = "gaussian", start = nullmod$varComp, 
-                                 AIREML.tol = AIREML.tol, max.iter = max.iter, EM.iter = EM.iter, 
+                                 group.idx = group.idx, family = "gaussian", start = nullmod$varComp,
+                                 AIREML.tol = AIREML.tol, max.iter = max.iter, EM.iter = EM.iter,
                                  drop.zeros = TRUE, verbose = verbose)
 
     ## add any extra slots
     extra <- setdiff(names(nullmod), names(new.nullmod))
     new.nullmod <- c(new.nullmod, nullmod[extra])
+    # Add sample id.
+    new.nullmod$fit$sample.id <- nullmod$fit$sample.id
+    # Update model string (but keep outcome the same?).
+    previous_model <- new.nullmod$model
+    # What happens if covMatList is different than what was passed to the original null model call?
+    new.nullmod$model <- paste(sprintf("rankInvNorm(resid(%s))", new.nullmod$outcome), "~",
+                               strsplit(previous_model, " ~ ")[[1]][2])
```

Comment:
The model string will be incorrect if the user passes a different set of covariance matrices when inverse-normal transforming the null model, but I can't see a valid reason that someone would do that (other than user error).

Reply:
See above comment - if model strings are usually handled in wrapper vs internal functions, would it make sense to move this to `nullModelInvNorm`?

Reply:
Agree, same comment as above.

Reply:
Oops, realized this one did have a wrapper - I ended up moving it to the wrapper.

Reply:
I moved the model string update lines to the wrapper.